tests/TrackJS/TrackJSEmbedTest.php

```diff
@@ -0,0 +1,134 @@
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $trackJsEmbed = new TrackJSEmbed(
+            $hostingConfig,
+            null,
+            null,
+            false
+        );
+
+        $this->expectException(AssetIntegrationFailedException::class);
+
+        $trackJsEmbed->getEmbedHtml();
+    }
+
+    public function testGetEmbedHtmlInDebugMode () : void
+    {
+        /** @var MockObject|AssetHelper $assetHelper */
+        $assetHelper = $this->getMockBuilder(AssetHelper::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        // region not in debug
+        /** @var MockObject|HostingConfig $hostingConfigEmptyToken */
+        $hostingConfigEmptyToken = $this->getMockBuilder(HostingConfig::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $hostingConfigEmptyToken->expects($this->atLeastOnce())
+            ->method("getTrackJsToken")
+            ->willReturn("thisIsAToken");
+
+        $hostingConfigEmptyToken
+            ->method("isInDevelopmentTier")
+            ->willReturn(false);
+
+        $trackJsEmbedEmptyToken = new TrackJSEmbed(
+            $hostingConfigEmptyToken,
+            $assetHelper,
+            null,
+            false
+        );
+
+        $this->assertNotEmpty($trackJsEmbedEmptyToken->getEmbedHtml());
+        // endregion
+
+
+        // region empty token
+        /** @var MockObject|HostingConfig $hostingConfigEmptyToken */
+        $hostingConfigEmptyToken = $this->getMockBuilder(HostingConfig::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $hostingConfigEmptyToken->expects($this->atLeastOnce())
+            ->method("getTrackJsToken")
+            ->willReturn(null);
+
+        $hostingConfigEmptyToken->method("isInDevelopmentTier")
+            ->willReturn(false);
+
+        $trackJsEmbedEmptyToken = new TrackJSEmbed(
+            $hostingConfigEmptyToken,
+            $assetHelper,
+            null,
+            false
+        );
+
+        $this->assertEmpty($trackJsEmbedEmptyToken->getEmbedHtml());
+        // endregion
+
+        // region isDebug
+        /** @var MockObject|HostingConfig $hostingConfigIsDebug */
+        $hostingConfigIsDebug = $this->getMockBuilder(HostingConfig::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $hostingConfigIsDebug->expects($this->atLeastOnce())
+            ->method("getTrackJsToken")
+            ->willReturn("thisIsAToken");
+
+        $hostingConfigIsDebug->method("isInDevelopmentTier")
+            ->willReturn(true);
+
+        $trackJsEmbedIsDebug = new TrackJSEmbed(
+            $hostingConfigIsDebug,
+            $assetHelper,
+            null,
+            true
+        );
+
+        $this->assertEmpty($trackJsEmbedIsDebug->getEmbedHtml());
+        // endregion
+
+        // region tier is not in deployment tiers
```

Comment:
The deployment tier here is not really used. You're just checking against `isInDevelopmentTier` and return `true`.

The description of the region doesn't really reflect what the test is doing, or am I missing something?